### PR TITLE
Document the new navigation intercepting method 

### LINKED ISSUE #229: The whole Navigation Interception section is outdated in docs
https://palindrom.github.io/docs/05-_Navigation_Interception/

Given the new development of `palindrom-before-redirect` and `palindrom-after-redirect` events, we need to re-write this page of docs around these events.

### PR DESCRIPTION
**Pull request by alshakero**
Fixes #229

### CODE REVIEW

**Comment by warpech** [comment]
Travis builds failed. I have restarted them

File: docs/05-_Navigation_Interception.md [hunk 1]
```diff
@@ -22,7 +22,39 @@ Example:
...
+-   `_top` - load the URL into the current browser tab or window
+-   `_blank` - load the URL into a new browser tab or window
 
 You can find more information on the `target` attribute at [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a).
+
+### Intercepting within-app navigation
+
+Before and after morphing the page upon navigation within your app, Palindrom dispatches bubbling navigation events. `palindrom-before-navigation` and `palindrom-after-navigation`. You can use these events to watch or even cancel navigation.
+
+To cancel Palindrom navigation, you can cancel `palindrom-before-navigation` event by calling `event.preventDefault()` inside your event handler. Then after you show your confirmation box or whatever prupose you canceled the navigation for, you can continue the navigation by calling `palindrom.morphUrl` to the same URL. To record the navigation URL, you can get it from `event.detail.href` of `palindrom-before-navigation`.
```
Comment (warpech) [inline docs/05-_Navigation_Interception.md]: prupose -> purpose

File: docs/05-_Navigation_Interception.md [hunk 2]
```diff
@@ -22,7 +22,39 @@ Example:
...
 As the value of the `target` attribute, you might consider one of the following keywords that have a special meaning in HTML:
 
-- `_top` - load the URL into the current browser tab or window
-- `_blank` - load the URL into a new browser tab or window
+-   `_top` - load the URL into the current browser tab or window
+-   `_blank` - load the URL into a new browser tab or window
 
 You can find more information on the `target` attribute at [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a).
+
+### Intercepting within-app navigation
```
Comment (warpech) [inline docs/05-_Navigation_Interception.md]: Perhaps change the `How to disable interception` header on line 11 to `How to disable interception for an HTML link`

And then change the header `Intercepting within-app navigation` to `Cancellable events before and after navigation`

File: docs/05-_Navigation_Interception.md [hunk 3]
```diff
@@ -22,7 +22,39 @@ Example:
...
+
+Before and after morphing the page upon navigation within your app, Palindrom dispatches bubbling navigation events. `palindrom-before-navigation` and `palindrom-after-navigation`. You can use these events to watch or even cancel navigation.
```
Comment (warpech) [inline docs/05-_Navigation_Interception.md]: `navigation events.` -> `navigation events,` (replace the dot with a comma)

File: docs/05-_Navigation_Interception.md [hunk 4]
```diff
@@ -22,7 +22,39 @@ Example:
...
+
+To cancel Palindrom navigation, you can cancel `palindrom-before-navigation` event by calling `event.preventDefault()` inside your event handler. Then after you show your confirmation box or whatever prupose you canceled the navigation for, you can continue the navigation by calling `palindrom.morphUrl` to the same URL. To record the navigation URL, you can get it from `event.detail.href` of `palindrom-before-navigation`.
+
+#### Note
+
+When you proceed with the navigation, since your new navigation is still a navigation, you'll trigger your event handlers again. To handle this you may to create a local flag and set it true after you handled the expected navigation.
```
Comment (warpech) [inline docs/05-_Navigation_Interception.md]: The first sentence is hard to understand :)

WDYT about changing this paragraph to:

> A programmatic call to `palindrom.morphUrl` also triggers the navigation events listed above. To distinguish between the events caused by a user interaction and a call to  `palindrom.morphUrl`, you may use a local variable.

File: docs/05-_Navigation_Interception.md [hunk 5]
```diff
@@ -22,7 +22,39 @@ Example:
...
+            // to show the dialog when navigation happens again
+            handledNavigation = false;
+        }
+    }
+});
+```
+
+### Intercepting external navigation
+
+When the users clicks a link that takes them to an external link, Palindrom does not interfere. You may use native [beforeunload](`https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event`) event.
```
Comment (warpech) [inline docs/05-_Navigation_Interception.md]: I would remove the header from line 58 and change the text to say:

Palindrom only triggers the events `palindrom-before-navigation` and `palindrom-after-navigation` for the intercepted links. For non-intercepted links, you may use the native [beforeunload](`https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event`) event.

**Comment by alshakero** [comment]
Ready for round II.

**Pull request merged**